plone/supermodel/model.py

```diff
@@ -107,10 +107,11 @@ def finalizeSchemas(parent=Schema):
         )
 
     def walk(schema):
-        yield schema
-        for child in schema.dependents.keys():
-            for s in walk(child):
-                yield s
+        if isinstance(schema, SchemaClass):
```

Comment:
What else (than a SchemaClass) did pop up here?

Reply:
I don't remember. Maybe this isn't needed with the latest Zope4 anymore, I'll have to check.
A similar check was removed back in 2016 (https://github.com/plone/plone.supermodel/commit/c1822bb822270b323558bfd3cd49229c50b04140), so maybe we should to a `raise TypeError` here and see what is failing.

Reply:
I agree to raise a `TypeError` here if `schema` is something different than a `SchemaClass`.